internal/wal_push_handler.go

```diff
@@ -75,9 +141,46 @@ func uploadWALFile(uploader *WalUploader, walFilePath string, preventWalOverwrit
 		return errors.Wrapf(err, "upload: could not open '%s'\n", walFilePath)
 	}
 	err = uploader.UploadWalFile(walFile)
+	if err == nil && (viper.IsSet(UploadWalMetadata) || viper.GetString(UploadWalMetadata) == WalNoMetadataLevel) {
+		err = uploadWALMetadataFile(uploader, walFilePath)
+		if err != nil {
+			return errors.Wrapf(err, "Failed to upload metadata file")
+		}
+	}
 	return errors.Wrapf(err, "upload: could not Upload '%s'\n", walFilePath)
 }
 
+// Function to upload WAL Metadata file based on the parameter passed
+func uploadWALMetadataFile(uploader *WalUploader, walFilePath string) error {
+	err := checkWalMetadataLevel(viper.GetString(UploadWalMetadata))
+	if err != nil {
+		return errors.Wrapf(err, "Incorrect wal metadta level")
+	}
+	fileStat, err := os.Stat(walFilePath)
+	if err != nil {
+		return errors.Wrapf(err, "upload: could not stat wal file'%s'\n", walFilePath)
+	}
+	var walMetadata WalMetadataDescription
+	walMetadataS := make(map[string]WalMetadataDescription)
```

Comment:
Rename walMetadataS to something more meaningful?

Reply:
Renamed it.